diff --git a/net/mosdns/Makefile b/net/mosdns/Makefile
index f2cdb6dff..40c0022f8 100644
--- a/net/mosdns/Makefile
+++ b/net/mosdns/Makefile
@@ -5,26 +5,21 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=mosdns
-PKG_VERSION:=5.1.1
-PKG_RELEASE:=1
+PKG_VERSION:=4.5.3
+PKG_RELEASE:=$(AUTORELEASE)
 
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
 PKG_SOURCE_URL:=https://codeload.github.com/IrineSistiana/mosdns/tar.gz/v$(PKG_VERSION)?
-PKG_HASH:=b184f4e3cb8b8017c4a7eb0742c3e80da54e46915e987b5d79e312da47fd751c
+PKG_HASH:=2a13b1b1d862731cf946a8ceaa99523a0eb0eaae56045dd31207b61a5a6d47ae
 
 PKG_LICENSE:=GPL-3.0
 PKG_LICENSE_FILES:=LICENSE
 PKG_MAINTAINER:=Tianling Shen <cnsztl@immortalwrt.org>
 
-PKG_BUILD_DEPENDS:=golang/host
 PKG_BUILD_PARALLEL:=1
 PKG_USE_MIPS16:=0
 
-GO_PKG:=github.com/IrineSistiana/mosdns
-GO_PKG_LDFLAGS_X:=main.version=v$(PKG_VERSION)
-
 include $(INCLUDE_DIR)/package.mk
-include ../../lang/golang/golang-package.mk
 
 define Package/mosdns
   SECTION:=net
@@ -36,11 +31,7 @@ define Package/mosdns
   PROVIDES:=mosdns-neo
 endef
 
-GO_PKG_TARGET_VARS:=$(filter-out CGO_ENABLED=%,$(GO_PKG_TARGET_VARS)) CGO_ENABLED=0
-
 define Package/mosdns/install
-	$(call GoPackage/Package/Install/Bin,$(1))
-
 	$(INSTALL_DIR) $(1)/etc/init.d
 	$(INSTALL_BIN) $(PKG_BUILD_DIR)/scripts/openwrt/mosdns-init-openwrt $(1)/etc/init.d/mosdns
 
@@ -48,16 +39,10 @@ define Package/mosdns/install
 	$(INSTALL_DATA) ./files/config.yaml $(1)/etc/mosdns/config.yaml
 endef
 
-define Package/mosdns/postinst
-#!/bin/sh
+define Build/Configure
+endef
 
-# check if we are on real system
-if [ -z "$${IPKG_INSTROOT}" ]; then
-	/etc/init.d/mosdns stop >/dev/null 2>&1
-	/etc/init.d/mosdns disable >/dev/null 2>&1
-fi
-exit 0
+define Build/Compile
 endef
 
-$(eval $(call GoBinPackage,mosdns))
 $(eval $(call BuildPackage,mosdns))
