diff --git a/target/linux/mediatek/Makefile b/target/linux/mediatek/Makefile
index 77930ed..e0d8766 100644
--- a/target/linux/mediatek/Makefile
+++ b/target/linux/mediatek/Makefile
@@ -14,6 +14,6 @@ KERNEL_TESTING_PATCHVER:=5.4
 include $(INCLUDE_DIR)/target.mk
 DEFAULT_PACKAGES += \
 	kmod-leds-gpio kmod-gpio-button-hotplug \
-	autocore-arm uboot-envtool luci-app-wrtbwmon
+	autocore-arm uboot-envtools
 
 $(eval $(call BuildTarget))
